plot_data.Rmd

```diff
@@ -0,0 +1,295 @@
+
+```{r}
+most_pop_song_each_year |>
+  ggplot(aes(x = year, y = track.popularity)) +
+  geom_point(color = "orange") +
+  theme(axis.text.x = element_text(angle = 40,hjust = 1)) +
+  labs(title = "Number one most popular song in each year",
+       subtitle = "From 1978-2010",
+       x = "Year",
+       y = "popularity") +
+ theme(panel.background = element_rect(fill = "black"),
+        panel.grid.major = element_line(color = "black"),
+        panel.grid.minor = element_line(color = "black"))
+```
+
+very unpopular songs having low ratings in popularity this playlist is filled with ear stredding ear bleeding horrible covers of songs
+
+
+First looks at the trash_songs dataset
+
+```{r}
+trash_songs |>
+  ggplot(aes(x = track.name, y = track.popularity)) +
+  geom_point()
+
+trash_songs |>
+  ggplot(aes(x = track.name, y = track.popularity)) +
+  geom_bar(stat = "identity")
+
+trash_songs |>
+  ggplot(aes(x = track.name, y = track.popularity)) +
+  geom_boxplot()
+```
+
+running t test between the popular songs and the unpopular songs
+
+```{r}
+t.test(trash_songs$track.popularity,top_songs_df$track.popularity)
+```
+
+This shows that there is indeed a difference in popularity in our two song sets. This helps validate our approach of using these two song sets to examine what variables might determine song popularity.
+
+
+```{r}
+all_songs_df |>
+  ggplot(aes(x = song_group, y = track.popularity)) +
+  geom_violin()
+
+ggplot(data = all_songs_df, aes(x = song_group, y = danceability)) +
+  geom_violin()
+```
+
+The set of plots below are comparing the popular and the unpopular songs to see if any of them have any noticeable differences but there actually isn't much of a difference between unpopular and popular songs for any of these variables which could possibly mean that unpopular songs try to copy popular songs cadence and flow.
+
+```{r}
+all_songs_df |>
+  ggplot(aes(x = song_group, y = danceability)) +
+  geom_violin()
+
+all_songs_df |>
+  ggplot(aes(x = song_group, y = tempo)) +
+  geom_violin()
+
+all_songs_df |>
+  ggplot(aes(x = song_group, y = liveness)) +
+  geom_violin()
+
+all_songs_df |>
+  ggplot(aes(x = song_group, y = instrumentalness)) +
+  geom_violin()
+
+all_songs_df |>
+  ggplot(aes(x = song_group, y = energy)) +
+  geom_violin()
+
+all_songs_df |>
+  ggplot(aes(x = song_group, y = loudness)) +
+  geom_violin()
+
+all_songs_df |>
+  ggplot(aes(x = song_group, y = acousticness)) +
+  geom_violin()
+
+all_songs_df |>
+  ggplot(aes(x = song_group, y = key)) +
+  geom_violin()
+
+## Now you can clean up trash_songs_df and make plots from it
+
+
+```
+
+
+```{r}
+other_bad_playlist <- other_bad_playlist |>
+  bind_rows() |>
+  write_tsv(file.path("data","other_bad_playlist.tsv"))
+
+another__bad_playlist <- another__bad_playlist |>
+  bind_rows() |>
+  write_tsv(file.path("data","another_bad_playlist.tsv"))
+```
+
+join both new datasets. 
+
+nbp_combined stands for new bad playlist combined
```

Comment:
It can be nice to use backticks to make places where you are talking about code stick out
```suggestion
`nbp_combined` stands for new bad playlist combined
```